lib/widgets/message_list.dart

```diff
@@ -785,66 +785,72 @@ class MessageWithPossibleSender extends StatelessWidget {
-            ).merge(weightVariableTextStyle(context))),
-        ]);
+      senderWidget = Padding(
+        padding: const EdgeInsets.only(top: 4, left: 16, right: 16),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          crossAxisAlignment: CrossAxisAlignment.baseline,
+          textBaseline: TextBaseline.alphabetic,
+          children: [
+            Flexible(
+              child: GestureDetector(
+                onTap: () => Navigator.push(context,
+                  ProfilePage.buildRoute(context: context,
+                    userId: message.senderId)),
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 2),
+                  child: Row(
+                    children: [
+                      Avatar(size: 32, borderRadius: 3, userId: message.senderId),
+                      const SizedBox(width: 8),
+                      Flexible(
+                        child: Text(message.senderFullName, // TODO get from user data
+                          style: const TextStyle(
+                            fontFamily: 'Source Sans 3',
+                            fontSize: 18,
+                            height: (22 / 18),
+                          ).merge(weightVariableTextStyle(context, wght: 600,
+                              wghtIfPlatformRequestsBold: 900)),
+                          overflow: TextOverflow.ellipsis),
+                      ),
+                    ]),
+                )),
+            ),
+            Text(time,
+              style: TextStyle(
+                color: _kMessageTimestampColor,
+                fontFamily: 'Source Sans 3',
+                fontSize: 16,
+                height: (18 / 16),
+                fontFeatures: const [FontFeature.enable('c2sc'), FontFeature.enable('smcp')],
+              ).merge(weightVariableTextStyle(context))),
+          ]));
     } else {
       senderWidget = const SizedBox.shrink();
     }
 
     return GestureDetector(
       behavior: HitTestBehavior.translucent,
       onLongPress: () => showMessageActionSheet(context: context, message: message),
-      child: Padding(
-        padding: const EdgeInsets.symmetric(vertical: 4),
-        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
-          const SizedBox(width: 16),
-          Expanded(
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: [
-                senderWidget,
-                MessageContent(message: message, content: item.content),
-                if ((message.reactions?.total ?? 0) > 0)
-                  ReactionChipsList(messageId: message.id, reactions: message.reactions!)
-              ])),
-          const SizedBox(width: 16),
-        ])));
+      child: Column(
+        children: [
+          senderWidget,
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4),
```

Comment:
I agree, and I think those extra 4px don't correspond to anything in the Figma design.

AFAICT this last commit is meant to have no effect on the visible layout. Is that right? I'm not sure the effect on the source code is an improvement; so perhaps that commit can just be left out. If something like it is needed for a followup change like indicating starred messages, then it can appear as a prep commit (but with those extra 4px fixed) in that PR, where it'll be easier to assess what it's doing for us.

Reply:
From: https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=134%3A8312&mode=dev
![messages_4px_padding](https://github.com/zulip/zulip-flutter/assets/98299/45daf2bf-ea59-4b9b-9ebd-5f985c0c1394)

The mockup here shows each distinct message item has 4px vertical margins. The sender widget is complicated in the design as it has 2px of vertical padding (making a total of 6px top margin if the sender is shown) and there is a 2px margin above the message content and the sender:

![message_avatar_content_padding](https://github.com/zulip/zulip-flutter/assets/98299/7d958261-fe45-4e1d-ad59-a99587a191fd)

So taking the figma as is, message contents would have dynamic padding depending if the sender is visible or not.  But I think to simplify things, given there is effectively 4px spacing between the contents and the sender (2px coming from the sender padding and 2px between a message content and the sender), there ends up being 4px vertical padding for message contents in all instances. If we then give the sender 6px top margin we achieve the same layout.

An alternative would be to put that 4px padding around the entire `MessageWithPossibleSender` (then giving the sender widget 2px top and 4px bottom padding), but my intention here with the layout is to leave the left/right gutters as space to add decorations (like the incoming star-marker) and having the padding in the container would limit how we could place such decorations, for example we could not align anything to the corner. There's no need for this currently but I do think a layout that gives us more flexibility to adjust those decorations is better.

(This discussion ignores that bottom padding on messages are expanded when it is before a new recipient header, but that case is already handled by introducing extra space in those situations)

Reply:
> So taking the figma as is, message contents would have dynamic padding depending if the sender is visible or not. But I think to simplify things, given there is effectively 4px spacing between the contents and the sender (2px coming from the sender padding and 2px between a message content and the sender), there ends up being 4px vertical padding for message contents in all instances. If we then give the sender 6px top margin we achieve the same layout.

Cool. Yeah, in general it's fine for the layout to be expressed differently from how it is in Figma as long as it produces an equivalent result. We've done that in a fair number of places already. For us the Figma design expresses a visual result, but not the details of how to implement it.